src/app/tasks/task.service.ts

```diff
@@ -40,59 +42,62 @@ export class TaskService {
     }
 
     private convertBasicTaskToTask(basicTask: BasicTask) : Task | null {
-        let taskIndex: number = this.tasksSubject.getValue().findIndex(task => task.id == basicTask.id);
+        let taskIndex: number = this.tasks.findIndex(task => task.id == basicTask.id);
         if(taskIndex === -1) return null;
-        return this.tasksSubject.getValue()[taskIndex];
+        return this.tasks[taskIndex];
     }
 
     private convertTasksToBasicTasks(tasks: Task[]) : BasicTask[] {
         let basicTasks : BasicTask[] = [];
-        this.tasksSubject.getValue().forEach(task => basicTasks.push(this.convertTaskToBasicTask(task)));
+        this.tasks.forEach(task => basicTasks.push(this.convertTaskToBasicTask(task)));
         return basicTasks;
     }
 
     public getTasks() : BasicTask[] {
-        return this.convertTasksToBasicTasks(this.tasksSubject.getValue()).slice()
+        return this.convertTasksToBasicTasks(this.tasks).slice()
         .sort(
             (taskA, taskB) => {
                 return (taskA.title < taskB.title) ? -1 : (taskA.title > taskB.title) ? 1 : 0
         });
     }
 
-    public getTask(taskId: number): Task | null {
-        return this.tasksSubject.getValue()?.find(task => {
-            return task.id == taskId;
-        }) ?? null;
+    public getTask(taskId: number): Observable<Task | null> {
+        return this.httpClient.get<Task | null>(beApiRoutes.taskDetails + taskId);
     }
 
     public addTask(task: Task) {
-        task.id = this.idGeneratorService.generateId();
-        this.tasksSubject.getValue().push(task);
+        let oldTaskList = this.tasks.slice();
+        this.httpClient.post<Task | null>(beApiRoutes.tasks, task)
+        .subscribe(createdTask => {
```

Comment:
don't subscribe here, you can use pipe and tap operators and subscribe to this method in the component